azuredevops/internal/service/approvalsandchecks/resource_check_required_template.go

```diff
@@ -0,0 +1,104 @@
+package approvalsandchecks
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/utils/pipelineschecksextras"
+)
+
+var validRepositoryTypes = []string{"git", "github", "bitbucket"}
+
+// ResourceCheckRequiredTemplate schema and implementation for required template check resources
+func ResourceCheckRequiredTemplate() *schema.Resource {
+	r := genBaseCheckResource(flattenCheckRequiredTemplate, expandCheckRequiredTemplate)
+
+	r.Schema["required_template"] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"repository_type": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					Default:      "git",
+					ValidateFunc: validation.StringInSlice(validRepositoryTypes, false),
+				},
+				"repository_name": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+				"repository_ref": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+				"template_path": {
```

Comment:
`template_path` should be `Required: true` not `Optional: true,`

Reply:
fixed